crates/turborepo-lib/src/run/summary/spaces.rs

```diff
@@ -62,10 +181,69 @@ impl SpacesClient {
             space_id,
             api_client,
             api_auth,
+            request_timeout: Duration::from_secs(10),
+            errors: Vec::new(),
         })
     }
 
-    async fn create_run(_payload: SpacesRunPayload) {
-        todo!()
+    pub async fn start(
+        mut self,
+        create_run_payload: CreateSpaceRunPayload<'_>,
+    ) -> Result<SpacesClientHandle, Error> {
+        let (tx, mut rx) = tokio::sync::mpsc::channel(100);
+        let run = self.create_run(create_run_payload).await?;
+
+        let handle = tokio::spawn(async move {
+            while let Some(req) = rx.recv().await {
+                let resp = match req {
+                    SpaceRequest::Completed { .. } => self.finish_run_handler(req, &run).await,
+                    SpaceRequest::Task { summary } => {
+                        self.finish_task_handler(*summary, &run).await
+                    }
+                };
+
+                if let Err(e) = resp {
+                    self.errors.push(e);
+                }
+            }
+
+            SpacesClientResult {
+                errors: self.errors,
+                run: Some(run),
+            }
+        });
+
+        Ok(SpacesClientHandle { handle, tx })
+    }
```

Comment:
I'm pretty sure this isn't what we want. Making this `async` means that upon creation we aren't moving forward with execution until that first spaces API request finishes which isn't what the Go code did. We want to be able to fire off the `create_run` request in the background and proceed to start execution ASAP.